pkg/storage/replica_range_lease.go

```diff
@@ -194,6 +194,14 @@ func (p *pendingLeaseRequest) requestLeaseAsync(
 					}
 				} else if status.Liveness.Epoch == *status.Lease.Epoch {
 					// If not owner, increment epoch if necessary to invalidate lease.
+					// However, we only do so in the event that the next leaseholder is
+					// considered live at this time. If not, there's no sense in
+					// incrementing the expired leaseholder's epoch.
+					if live, liveErr := repl.store.cfg.NodeLiveness.IsLive(nextLeaseHolder.NodeID); !live || liveErr != nil {
```

Comment:
I like this idea, but it looks like you failed to actually skip incrementing the epoch if this condition is true. I'm rerunning a couple experiments and have fixed this logic up locally to actually return here.